mobify/mobify-code-style
# Copyright Headers Tool

## Changes
- This PR adds a tool to either check for existing copyright headers, or add copyright headers if they are missing
- This tool is language agnostic (see note below), and can be pointed to any directory using a glob pattern


**Note:** this tool reads from the `headers/copyright.extension` files to select what header to insert, where `extension` is the target language

Currently only `.py`, `.jsx`, and `.js` have been added for testing.

## How To Test

- Clone the repo and checkout this branch (`copyright-headers`)
- `cd` into the `copyright` directory, and run `node copyright.js 'PATTERN'` where `PATTERN` is a glob pattern
- For example, if you cloned this repo in the same directory as the `progressive-web-sdk`, and needed to add copyright headers to all `.js` files in the `src` directory of the SDK, you would write:

```bash
node copyright.js '../../progressive-web-sdk/src/**/*.js' --fix
```

Note the optional `--fix` flag that can be passed in, which will add the headers to files which do not contain it.

Another flag `--update` will check the target files for outdated headers, and update them to the current year.

The default mode (no flags) will exit the process with an error if any of the targeted files are missing copyright headers.


### To test in dev mode (with chrome inspect)
`node --inspect --debug-brk copyright.js ../../progressive-web-sdk/bin/**/*.js`


## TODOs:


## Resources:
[node-glob](https://github.com/isaacs/node-glob)
